sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigquery/FileLoadsStreamingIT.java

```diff
@@ -0,0 +1,511 @@
+                .startAt(start)
+                .stopAt(start.plus(Duration.standardSeconds(TOTAL_N - 1)))
+                .withInterval(Duration.standardSeconds(1))
+                .catchUpToNow(false));
+    PCollection<TableRow> rows =
+        instants.apply(
+            "Create TableRows",
+            MapElements.into(TypeDescriptor.of(TableRow.class))
+                .via(instant -> generateRowFunc.apply(instant.getMillis() / 1000)));
+    // build write transform
+    Write<TableRow> write =
+        BigQueryIO.writeTableRows()
+            .to(tableSpec)
+            .withMethod(Write.Method.FILE_LOADS)
+            .withTriggeringFrequency(Duration.standardSeconds(10));
+    if (useCopyJobs) {
+      write = write.withMaxBytesPerPartition(250);
+    }
+    if (useInputSchema) {
+      // we're creating the table with the input schema
+      write =
+          write
+              .withSchema(inputSchema)
+              .withCreateDisposition(CreateDisposition.CREATE_IF_NEEDED)
+              .withWriteDisposition(WriteDisposition.WRITE_TRUNCATE);
+    } else {
+      // table already exists with a schema, no need to create it
+      write =
+          write
+              .withCreateDisposition(CreateDisposition.CREATE_NEVER)
+              .withWriteDisposition(WriteDisposition.WRITE_APPEND);
+    }
+    write = numFileShards == 0 ? write.withAutoSharding() : write.withNumFileShards(numFileShards);
+
+    rows.apply("Stream loads to BigQuery", write);
+    p.run().waitUntilFinish();
+
+    List<TableRow> expectedRows = new ArrayList<>();
+    for (long i = 0; i < TOTAL_N; i++) {
+      expectedRows.add(generateRowFunc.apply(i));
+    }
+
+    // Perform checks
+    checkRowCompleteness(tableSpec, inputSchema, expectedRows);
+  }
+
+  // Check that the expected rows reached the table.
+  private static void checkRowCompleteness(
+      String tableSpec, TableSchema schema, List<TableRow> expectedRows)
+      throws IOException, InterruptedException {
+    List<TableRow> actualTableRows =
+        BQ_CLIENT.queryUnflattened(
+            String.format("SELECT * FROM [%s]", tableSpec), PROJECT, true, false);
+
+    Schema rowSchema = BigQueryUtils.fromTableSchema(schema);
+    List<Row> actualBeamRows =
+        actualTableRows.stream()
+            .map(tableRow -> BigQueryUtils.toBeamRow(rowSchema, tableRow))
+            .collect(Collectors.toList());
+    List<Row> expectedBeamRows =
+        expectedRows.stream()
+            .map(tableRow -> BigQueryUtils.toBeamRow(rowSchema, tableRow))
+            .collect(Collectors.toList());
+    LOG.info(
+        "Actual rows number: {}, expected: {}", actualBeamRows.size(), expectedBeamRows.size());
+
+    assertThat(
+        "Comparing expected rows with actual rows",
+        actualBeamRows,
+        containsInAnyOrder(expectedBeamRows.toArray()));
+    assertEquals(
+        "Checking there is no duplication", expectedBeamRows.size(), actualBeamRows.size());
+  }
+
+  @Test
```

Comment:
It may not be necessary to run a full matrix of every variable (fixed shard, dynamical destination,  use copyJobs) which generates 2*2*2 possibilities.

A choice to reduce the complexity is test default + each flag in a test + test all feature added, this reduces the complexity from 2^n to n+2. Also, given the default case (had autoshard) will be tested in #28272 (serve as default), I propose keep the following one:

- testLoadWithFixedShards
- testWithAutoShardingAndCopyJobs
- testDynamicDestinationsWithAutoSharding
- testDynamicDestinationsWithAutoShardingAndCopyJobs

Reply:
P.S. I think we can remove the FILE_LOADS test in #28272 and keep FileLoadsStreamingIT as the hub for streaming loads tests. That way we can keep the other file as `BigQueryIOStorageWriteIT` for storage write. wdyt?

Reply:
> we should do testDynamicDestinationsWithFixedShards instead of testDynamicDestinationsWithAutoSharding

Sure that's good